File: Greedy/LET_no452_Minimum Number of Arrows to Burst Balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points = sorted(points, key = lambda x: x[1])
        res, end = 0, -sys.maxsize
        
        for left, right in points:
            if left > end:
                res += 1
                end = right
        
        return res

File: Greedy/LET_no452_Minimum Number of Arrows to Burst Balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        if (points.size() == 0)
            return 0;
        
        sort(points.begin(), points.end(), [](vector<int>& a, vector<int>& b) { return a[1] < b[1]; });
        int res = 1, end = points[0][1];

        for (vector<int>& p : points) {
            int left = p[0], right = p[1];
            if (left > end) {
                res++;
                end = right;
            }
        }

        return res;
    }
};